packages/react-aria-components/stories/Autocomplete.stories.tsx

```diff
@@ -856,3 +860,268 @@ export const AutocompleteSelect = (): React.ReactElement => (
+                <Cell>Games</Cell>
+                <Cell>File folder</Cell>
+                <Cell>6/7/2020</Cell>
+              </Row>
+              <Row id="2" style={{width: 'inherit', height: 'inherit'}}>
+                <Cell>
+                  <MyCheckbox slot="selection" />
+                </Cell>
+                <Cell>Program Files</Cell>
+                <Cell>File folder</Cell>
+                <Cell>4/7/2021</Cell>
+              </Row>
+              <Row id="3" style={{width: 'inherit', height: 'inherit'}}>
+                <Cell>
+                  <MyCheckbox slot="selection" />
+                </Cell>
+                <Cell>bootmgr</Cell>
+                <Cell>System file</Cell>
+                <Cell>11/20/2010</Cell>
+              </Row>
+              <Row id="4" style={{width: 'inherit', height: 'inherit'}}>
+                <Cell>
+                  <MyCheckbox slot="selection" />
+                </Cell>
+                <Cell>log.txt</Cell>
+                <Cell>Text Document</Cell>
+                <Cell>1/18/2016</Cell>
+              </Row>
+            </TableBody>
+          </Table>
+        </Virtualizer>
+      </div>
+    </AutocompleteWrapper>
+  );
+};
+
+export const AutocompleteWithTagGroup = () => {
+  return (
+    <AutocompleteWrapper>
+      <div>
+        <TextField autoFocus data-testid="autocomplete-example">
+          <Label style={{display: 'block'}}>Test</Label>
+          <Input />
+        </TextField>
+        <TagGroup>
+          <Label>Categories</Label>
+          <TagList style={{display: 'flex', gap: 4}}>
+            <MyTag href="https://nytimes.com">News</MyTag>
+            <MyTag>Travel</MyTag>
+            <MyTag>Gaming</MyTag>
+            <TooltipTrigger>
+              <MyTag>Shopping</MyTag>
+              <Tooltip
+                offset={5}
+                style={{
+                  background: 'Canvas',
+                  color: 'CanvasText',
+                  border: '1px solid gray',
+                  padding: 5,
+                  borderRadius: 4
+                }}>
+                <OverlayArrow style={{transform: 'translateX(-50%)'}}>
+                  <svg width="8" height="8" style={{display: 'block'}}>
+                    <path d="M0 0L4 4L8 0" fill="white" strokeWidth={1} stroke="gray" />
+                  </svg>
+                </OverlayArrow>
+                I am a tooltip
+              </Tooltip>
+            </TooltipTrigger>
+          </TagList>
+        </TagGroup>
+      </div>
+    </AutocompleteWrapper>
+  );
+};
+
+function AutocompleteNodeFiltering(args) {
+  let {contains} = useFilter({sensitivity: 'base'});
+  let filter = (textValue, inputValue, node) => {
+    if ((node.parentKey === 'Section 1' && textValue === 'Open View') || (node.parentKey === 'Section 2' && textValue === 'Appearance')) {
+      return true;
+    }
+    return contains(textValue, inputValue);
+  };
+
+  return (
+    <Autocomplete filter={filter}>
```

Comment:
for instance, is this picking things up correctly? I would've thought that you'd need to specify the `<Autocomplete<{name: string, isSection?: boolean, children?: ReactNode}>` to match the types of `items`
I don't think Autocomplete will be able to infer this, so I would expect the generic to be required. Then `filter` can be typed as well.